packages/builder/src/builderStore/store/frontend.js

```diff
@@ -490,6 +541,15 @@ export const getFrontendStore = () => {
           }
         })
 
+        // Set a default datasource
+        if (dataSourceField && defaultDatasource) {
+          props["dataSource"] = {
```

Comment:
We can't assume the name of the setting is `dataSource` here - we should reference the `key` property of the datasource/table setting found earlier.